packages/ember-template-compiler/lib/plugins/utils.ts

```diff
@@ -12,25 +12,41 @@ export function isStringLiteral(node: AST.Expression): node is AST.StringLiteral
   return node.type === 'StringLiteral';
 }
 
+function getLocalName(node: string | AST.Expression | AST.VarHead) {
+  if (typeof node === 'string') return node;
+
+  if (node.type === 'VarHead') {
+    return node.original;
+  }
+
+  // surely this is wrong? (does it ever occur tho?)
+  // the type of params is `| Expression`, which isn't possible for block params
+  return node;
+}
+
 export function trackLocals() {
   let locals = new Map();
 
   let node = {
-    enter(node: AST.Program | AST.Block | AST.ElementNode) {
-      for (let param of node.blockParams) {
+    enter(node: AST.Template | AST.Block | AST.ElementNode | AST.BlockStatement) {
```

Comment:
You don't want `AST.BlockStatement` here. `BlockStatement` is the `{{#foo}}...{{/foo}}`, the block params is on the default/inverse blocks (which should get visited by the `Block` visitor). 

There _is_ a `params` on `BlockStatement` but it means a completely different thing, it is the positional argument for the invocation, which has a type of `AST.Expression[]` hence you question above.

On the other hand, `node.params` on `AST.Template | AST.Block | AST.ElementNode` should all be `AST.VarHead[]`.